### DISCUSSION TITLE
Slowness with large asset numbers

### DISCUSSION THREAD

**Discussion by kmcluckie** [statamic/cms#5209] [Troubleshooting] [answered]
I have a site that also uses a database and has a public folder with over 30,000 images in it. I am not interested in accessing those images in Statamic. However, because those images are present in the "public" asset container, it slows down all use of that container.

Is it possible to prevent these images from being "processed" by Statamic? At this point I am not sure what is actually causing the slowdown, but even when I am not looking in the folder with the images themselves (for instance, just the 3 folders in the root of the public container) the slowness manifests. Any direction on this would be appreciated - I cut a few seconds off the load time by disabling the stache indexing, but the load time is still way too long to be useable.

Much thanks.

**Comment by jackmcdade** [comment]
Are these images in an **asset container**? If so, they're being scanned and a cache of the meta data is being stored on them. If you never need them processed by Statamic (e.g. available in an asset picker in the CP, etc), you can repoint your asset container to a different location and Statamic will not have any idea they exist and it shouldn't affect a thing.

**Comment by kmcluckie** [comment]
Yes, they are in an asset container.

Perhaps I don't understand, but since I need these images and my manually uploaded images (that I will use in Statamic) to be publicly visible, they both need to be on the public disk. So these thousands of images end up being in the same container, and thus indexed.

**Reply by jackmcdade** [reply]
Do you need all 30k of them to be selectable in the CP? Anything that doesn't need to be picked in the CP can still be in a public container, but an **asset** container is special. You could probably have two different locations and work it out.

**Comment by kmcluckie** [comment]
Ah, I may be able to make secondary public disk? I will experiment with that.

**Comment by jasonvarga** [comment]
Turning off the stache watcher should speed things up, too.

https://github.com/statamic/statamic/blob/eb4fb210205428de3291bfb2f9260973955fddfc/config/statamic/stache.php#L16

**Comment by kmcluckie** [comment, marked as answer]
I had already moved the stache watcher and that gave me some benefit, but it was still very slow.

I have now moved those images to a second public disk, and everything is nice and fast now. Thanks for the help.
